File: analytics/testkitchen/src/test/java/org/wikimedia/testkitchen/config/SampleConfigTest.java
package org.wikimedia.testkitchen.config;

import static org.assertj.core.api.Assertions.assertThat;
import static org.wikimedia.testkitchen.config.sampling.SampleConfig.Identifier.DEVICE;

import org.junit.jupiter.api.Test;
import org.wikimedia.testkitchen.config.sampling.SampleConfig;
import org.wikimedia.testkitchen.json.GsonHelper;

import com.google.gson.Gson;

class SampleConfigTest {

    @Test void testSamplingConfigDeserialization() {
        Gson gson = GsonHelper.getGson();
        String samplingConfigJson = "{\"rate\":0.25,\"identifier\":\"device\"}";
        SampleConfig sampleConfig = gson.fromJson(samplingConfigJson, SampleConfig.class);
        assertThat(sampleConfig.getRate()).isEqualTo(0.25);
        assertThat(sampleConfig.getIdentifier()).isEqualTo(DEVICE);
    }

}

File: analytics/testkitchen/src/test/java/org/wikimedia/testkitchen/config/SampleConfigTest.kt
package org.wikimedia.testkitchen.config

import org.junit.Assert.assertEquals
import org.junit.Test
import org.wikimedia.testkitchen.JsonUtil
import org.wikimedia.testkitchen.config.sampling.SampleConfig

internal class SampleConfigTest {
    @Test
    fun testSamplingConfigDeserialization() {
        val samplingConfigJson = "{\"rate\":0.25,\"unit\":\"device\"}"
        val sampleConfig = JsonUtil.decodeFromString<SampleConfig>(samplingConfigJson)!!
        assertEquals(0.25, sampleConfig.rate, 0.0000001)
        assertEquals(SampleConfig.UNIT_DEVICE, sampleConfig.unit)
    }
}
